core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -165,6 +172,31 @@ public SqlStatements setSqlLogger(SqlLogger sqlLogger) {
         return this;
     }
 
+    @Beta
+    public Integer getQueryTimeout() {
+        return queryTimeout;
+    }
+
+    /**
+     * Jdbi does not implement its own timeout mechanism: it simply calls {@link java.sql.Statement#setQueryTimeout}, leaving timeout handling to your jdbc driver.
+     *
+     * @param seconds the time in seconds to wait for a query to complete; 0 to disable the timeout; null to leave it at defaults (i.e. Jdbi will not call {@code setQueryTimeout(int)})
+     */
+    @Beta
+    public SqlStatements setQueryTimeout(@Nullable Integer seconds) {
```

Comment:
Should we take a `Duration` instead of `Integer`?  Attaching units to values prevents many types of common errors (like, assuming this timeout is in milliseconds!)

Reply:
Yeah, an integer number of seconds is pretty dumb here... I would have picked millis :)

Reply:
I'll give you a few minutes to add that if you'd like, then will merge.  I agree that with seconds resolution only it's not nearly as compelling.

Reply:
I think adding an overload that takes a Duration would be misleading to users since anything below seconds will be rounded, and throwing an IAE to avoid the rounding will be just as much of a PITA :/